plugins/action/common/merge_defaults.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2024 Cisco Systems, Inc. and its affiliates
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+from ansible.utils.display import Display
+from ansible.plugins.action import ActionBase
+
+display = Display()
+
+
+def merge_dicts(d1, d2):
+    for k, v in d2.items():
+        if k in d1 and isinstance(d1[k], dict) and isinstance(v, dict):
+            d1[k] = merge_dicts(d1[k], v)
+        else:
+            d1[k] = v
+    return d1
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        # self._supports_async = True
+        results = super(ActionModule, self).run(tmp, task_vars)
+        results['failed'] = False
+        results['msg'] = None
+        results['defaults'] = None
+
+        # Get Data from Ansible Task
+        fac_def = self._task.args['factory_defaults']
+        md = self._task.args['model_data']
+
+        cus_def = {}
+        if md is not None:
+            if md['defaults'] is not None:
```

Comment:
Please change this line to `if md.get('defaults') is not None:`

Otherwise we see the following error if defaults is not defined in host_vars

```
TASK [cisco.nac_dc_vxlan.validate : Merge factory and custom defaults] ***********************************************************************************************************************
task path: /home/cisco/Documents/nac-vxlan/collections/ansible_collections/cisco/nac_dc_vxlan/roles/validate/tasks/sub_main.yml:94
The full traceback is:
Traceback (most recent call last):
  File "/home/cisco/.pyenv/versions/nac-ndfc/lib/python3.11/site-packages/ansible/executor/task_executor.py", line 165, in run
    res = self._execute()
          ^^^^^^^^^^^^^^^
  File "/home/cisco/.pyenv/versions/nac-ndfc/lib/python3.11/site-packages/ansible/executor/task_executor.py", line 637, in _execute
    result = self._handler.run(task_vars=vars_copy)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/cisco/Documents/nac-vxlan/collections/ansible_collections/cisco/nac_dc_vxlan/plugins/action/common/merge_defaults.py", line 57, in run
    if md['defaults'] is not None:
       ~~^^^^^^^^^^^^
KeyError: 'defaults'
fatal: [nac-ndfc1 -> localhost]: FAILED! => {
    "msg": "Unexpected failure during module execution: 'defaults'",
    "stdout": ""
}
```